api/v1alpha1/shared_types.go

```diff
@@ -64,12 +64,6 @@ const (
 	APISchemaGCPAnthropic APISchema = "GCPAnthropic"
 )
 
-const (
-	// AIModelHeaderKey is the header key whose value is extracted from the request by the ai-gateway.
-	// This can be used to describe the routing behavior in HTTPRoute referenced by AIGatewayRoute.
-	AIModelHeaderKey = "x-ai-eg-model"
```

Comment:
This is not really an "internal" as it's used in the user-defined AIGatewayRules. For example, some user (I assume BB) programmatically generates the AIGatewayRoute where they define the rule using this constant. In other words, this could be also an breaking change, though not that serious it's not like k8s will be broken.

So I would suggest that keep this and make the alias in internalapi where we document for the internal codebase for this PR. wdyt?

Reply:
yeah good point. I didn't notice this was not internal